fs-dir-common.go

```diff
@@ -140,7 +141,7 @@ func treeWalk(bucketDir, prefixDir, entryPrefixMatch, marker string, recursive b
 	}
 
 	// scandir returns entries that begins with entryPrefixMatch
-	dirents, err := scandir(filepath.Join(bucketDir, prefixDir), prefixMatchFn, true)
+	dirents, err := scandir(path.Join(bucketDir, prefixDir), prefixMatchFn, true)
```

Comment:
Depending on the actual need to use `filepath.FromSlash` in `filepath.Join`, we might have to add it here too.